Add Interest feature

radl97 opened the pull request:
Adds a UI element tied back to the CircleEntity. Anyone can increment this counter. (any number of times)

The basic idea is to show interest in a possible opening for motivating the provider.

Note: Probably this needs migrations and etc. Also I could not test the code. I am happy to help, and discuss ideas :)

Gerviba commented on src/main/kotlin/hu/kirdev/schpincer/model/CircleEntity.kt:
```diff
@@ -79,6 +79,9 @@ data class CircleEntity(
         @Column
         var virGroupId: Long? = null
 
+        @Column
```
It requires a column definistion with '... default 0' parameter in order to make DB migration automatic

radl97 replied:
I've found this: https://stackoverflow.com/a/42150940 as an alternative to this: https://stackoverflow.com/a/375202
I think you were mentioning the latter.

However, can you check whether the former is valid? It seems a better solution, as you already use `org.hibernate`.

radl97 replied:
I've implemented the one using `org.hibernate`'s `ColumnDefault` for now

Gerviba commented on src/main/kotlin/hu/kirdev/schpincer/web/ApiController.kt:
```diff
@@ -336,6 +336,22 @@ open class ApiController(
 
+    @PostMapping("/provider/{circle}/increaseInterest")
+    @ResponseBody
+    fun circleInterestPlusPlus(@PathVariable circle: String, model: Model, request: HttpServletRequest): String {
```
This should be in a transaction

radl97 replied:
I'll pull the transactional part to service layer, as it is recommended (as I see it, and as it was pointed out here: https://stackoverflow.com/a/18637119 ).

radl97 replied:
(Done)

Gerviba commented on src/main/resources/templates/circleProfile.html:
```diff
@@ -116,6 +122,23 @@ <h4 th:text="*{name}">Full Name</h4>
     <div th:replace="MainLayout :: footer"></div>
+    <script>
+        function increaseInterest() {
+            return fetch("/api/provider/(Burgir)/increaseInterest", { // TODO hook in provider?
```
I might use sg. like:

```
function increaseInterest(provider) {
    return fetch(`/api/provider/${provider}/increaseInterest`, {
```

radl97 replied:
This gets really ugly (alias is injected as JavaScript from a template, and I do not like the sound of it :/ ). Or would it be okay? Or you had something else in mind?

For reference, this is how I understood your comment:
```patch
diff --git a/src/main/resources/templates/circleProfile.html b/src/main/resources/templates/circleProfile.html
index 7bc09ae..af6b985 100644
--- a/src/main/resources/templates/circleProfile.html
+++ b/src/main/resources/templates/circleProfile.html
@@ -58,7 +58,7 @@
                         <tr>
                             <td th:text="#{lang.profile.open}">Please open!</td>
                             <td>
-                                <button id="interest-counter" class="action-button" onclick="increaseInterest(1); return false" th:text="${selectedCircle?.interestCounter}" class="action-button">0</button>
+                                <button id="interest-counter" class="action-button" th:onclick="increaseInterest(${selectedCircle.alias}); return false" th:text="${selectedCircle?.interestCounter}" class="action-button">0</button>
                             </td>
                         </tr>
                         <tr>
@@ -123,8 +123,8 @@

     <div th:replace="MainLayout :: footer"></div>
     <script>
-        function increaseInterest() {
-            return fetch("/api/provider/(Burgir)/increaseInterest", { // TODO hook in provider?
+        function increaseInterest(provider) {
+            return fetch(`/api/provider/${provider}/increaseInterest`, {
                 method: 'POST',
                 cache: 'no-cache',
                 credentials: 'same-origin',
```

Gerviba replied:
In a normal scenario you could write sg. like:
```
th:onclick="|increaseInterest(${selectedCircle.alias}); return false|"
```
Notice the string template literal vertical bars.

But in case of th:onclick you are not able to use variables, so you have two options:
- use data atibutes: https://stackoverflow.com/a/54840163
- use inlined javascript template:
```
<script th:inline="javascript">
const provider = /*[[${selectedCircle.alias}]]*/ '';
</script>
```

radl97 replied:
I forgot to reflect the change in the comment.

radl97 replied:
I have used data attributes in the changes, can you check if it works?

radl97 replied:
for you :D

Gerviba reviewed (changes requested):
Thank you for your contribution!
I just reviewed your code, if you need help, contact me here or in pm.

radl97 commented:
Thank you for looking into this PR!
I think I've addressed all your comments.

Feel free to add nitpicks. I am not really confident with my Spring :D
